File: dataset/human/java/file_8747.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int a = sc.nextInt();
		int b = sc.nextInt();
		int k = sc.nextInt();

		int c = Math.min(a, b);
		int cnt = 0;
		int ans = 0;
		for (int i = c; i > 0; i--) {
			if (a % i == 0 && b % i == 0) {
				cnt++;
				if (cnt == k) {
					ans = i;
					break;
				}
			} else {
				continue;
			}
		}
		System.out.println(ans);
	}

}

File: dataset/human/c/file_8747.c
#include<stdio.h>
int main(void){
    int N, a[100001], i, j, ans;
    scanf("%d", &N);
    for(i = 1; i <= N; i++){
        scanf("%d", &a[i]);
    }
    ans = -1;
    j = 1;
    for(i = 1; i <= N; i++){
        if(a[j] == 2){
            ans = i;
            break;
        }else{
            j = a[j];
        }
    }
    printf("%d\n", ans);
    return 0;
}
